hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenUpdateNftsHandler.java

```diff
@@ -151,4 +171,37 @@ private void validateSemantics(
         }
         validateTrue(op.serialNumbers().size() <= tokensConfig.nftsMaxBatchSizeUpdate(), BATCH_SIZE_LIMIT_EXCEEDED);
     }
+
+    /**
+     * Creates a threshold key with threshold 1 with the given keys.
+     * @param keysRequired keys required
+     * @return threshold key with threshold 1
+     */
+    private Key oneOf(@NonNull final Key... keysRequired) {
+        return Key.newBuilder()
+                .thresholdKey(ThresholdKey.newBuilder()
+                        .keys(new KeyList(Arrays.asList(keysRequired)))
+                        .threshold(1)
+                        .build())
+                .build();
+    }
+
+    /**
+     * Verifies that all serial numbers are owned by the treasury account.
+     * @param treasuryAccount the treasury account
+     * @param serialNumbers the serial numbers
+     * @param nftStore the nft store
+     * @param tokenId the token id
+     * @return true if all serial numbers are owned by the treasury account
+     */
+    private boolean serialNumbersInTreasury(
+            @NonNull final AccountID treasuryAccount,
+            @NonNull final List<Long> serialNumbers,
+            @NonNull final ReadableNftStore nftStore,
+            @NonNull final TokenID tokenId) {
+        return serialNumbers.stream()
+                .map(serialNumber -> nftStore.get(tokenId, serialNumber))
```

Comment:
We should probably use `getIfUsable` here also. That has check for `INVALID_NFT_ID` check